test/test_nvfuser_frontend.py

```diff
@@ -0,0 +1,176 @@
+# Owner(s): ["module: nvfuser"]
+
+import unittest
+
+import torch
+from torch.testing._internal.common_utils import run_tests, TEST_WITH_ROCM, TestCase
+from torch.testing._internal.jit_utils import RUN_CUDA
+from torch._C._nvfuser import FusionManager, FusionDefinition, DataType
+import torch._refs as refs
+import torch._prims as prims
+
+RUN_NVFUSER = RUN_CUDA and not TEST_WITH_ROCM
+
+def is_pre_volta():
+    if not RUN_NVFUSER:
+        return False
+    prop = torch.cuda.get_device_properties(torch.cuda.current_device())
+    return prop.major < 7
+
+@unittest.skipIf(not RUN_NVFUSER, "requires CUDA")
+@unittest.skipIf(is_pre_volta(), "Only supported on Volta and newer devices.")
+class TestNVFuserFrontend(TestCase):
+    def test_basic(self) :
+        fm = FusionManager.get()
+        with FusionDefinition(fm) as fd :
+            t0 = fd.define_tensor(3)
+            t1 = fd.define_tensor(3)
+            c0 = fd.define_constant(3.0)
+
+            t2 = fd.ops.add(t0, t1)
+            t3 = fd.ops.mul(t2, c0)
+            t4 = fd.ops.sum(t3, [-1], False, DataType.Float)
+
+            fd.add_output(t4)
+
+        input1 = torch.ones(2, 4, 8, device='cuda')
+        input2 = torch.ones(2, 4, 8, device='cuda')
+
+        # Expected Output is a tensor of 48's
+        nvf_out1 = fm.execute([input1, input2])[0]
+
+        # Run the same definition to check caching
```

Comment:
IIUC, if you redefine the same fusion, then this will be a cache-hit, and no new `Fusion` object being created? Instead of just checking the output in this test, is there a way to test if this is actually a cache hit? For example, does it make sense to add a new method `FusionManager.num_fusions()` and assert that this number does not change after the definition of the second fusion?

Reply:
I changed the `FusionManager` to `FusionCache` and removed the need to use the singleton in the definition.  I also, added a check of `FusionCache.num_fusions()` in a python test to show the cache was hit since it did not change.